ui/component/or-dashboard-builder/src/widgets/or-image-widget.ts

```diff
@@ -0,0 +1,366 @@
+            </div>
+            <div>
+                ${this.expandedPanels.includes(i18next.t('image settings')) ? html`
+                    <div style="padding: 24px 24px 48px 24px;">
+                        <div>
+                        <input type="file" @change="${this.handleFileInputChange}" accept="${this.accept}"/> 
+                        </div>
+                    </div>
+                ` : null}
+            </div>
+        `
+        return output;
+    }
+
+    private prepareCoordinateEntries(config: ImageWidgetConfig, name: string){
+        var min = 0;
+        var max = 100;
+        if (config.attributeRefs && config.attributeRefs.length > 0) {
+            this.updateConfig(this.widget!, config);
+            console.log("Okay so I think we need to initialize the map here by why doesn't it work when we did earlier");
+            return config.attributeRefs.map((attr) => 
+            (html`
+                    <div style="margin: 5%; font-family: inherit; width: 100%;">${attr.name}</div>
+                    <or-mwc-input .type="${InputType.RANGE}" .min="${min}" .max="${max}" .value="${config.xCoordinatesMap[config.attributeRefs.indexOf(attr)]}"
+                    @or-mwc-input-changed="${(event: OrInputChangedEvent) => {
+                        config.xCoordinatesMap[config.attributeRefs.indexOf(attr)] = event.detail.value;
+                        this.updateConfig(this.widget!, config);
+                    }}"
+                    ></or-mwc-input>
+                    <or-mwc-input .type="${InputType.RANGE}" .min="${min}" .max="${max}" .value="${config.yCoordinatesMap[config.attributeRefs.indexOf(attr)]}"
+                    @or-mwc-input-changed="${(event: OrInputChangedEvent) => {
+                        config.yCoordinatesMap[config.attributeRefs.indexOf(attr)] = event.detail.value;
+                        this.updateConfig(this.widget!, config);
+                    }}"
+                    ></or-mwc-input>`));
+
+        }
+    }
+
+    private handleFileInputChange(event: Event) {
+        const config = JSON.parse(JSON.stringify(this.widget!.widgetConfig)) as ImageWidgetConfig;
+        const input = event.target as HTMLInputElement;
+        if (input.files && input.files[0]) {
+            
+            this._fileElem = input;
+            config.imagePath = URL.createObjectURL(input.files[0]);
+            this.updateConfig(this.widget!, config);
+            this.requestUpdate();
+        }
+    }
+
+    private updateConfig(widget: DashboardWidget, config: OrWidgetConfig | any, force: boolean = false) {
+        const oldWidget = JSON.parse(JSON.stringify(widget)) as DashboardWidget;
+        widget.widgetConfig = config;
+        this.requestUpdate("widget", oldWidget);
+        this.forceParentUpdate(new Map<string, any>([["widget", widget]]), force);
+    }
+
+    private onAttributesUpdate(changes: Map<string, any>) {
```

Comment:
Could imagine being confused by this function. I'll explain it's original use case:

It's basically meant for "when the list of selected attributes update, do something based on the changes".
So, as you did here, to update the `loadedAssets` if they have been changed.
But in `or-gauge-widget` and `or-kpi-widget` for example, we update the `displayName` of the widget when
the selected asset changes.
Things like that haha

So all good what you're doing here 👍